### ISSUE TITLE
[BUG] Parser fails when Twig syntax tokens appear inside Twig comments

### ISSUE THREAD

**Issue by webrgp** [yeonjuan/html-eslint#457]
The `@html-eslint/parser` throws a parsing error when Twig syntax delimiters (`{{`, `{%`) appear inside Twig comments (`{# #}`). The parser incorrectly interprets these nested tokens as the start of new template expressions instead of treating them as part of the comment content.

**Expected Behavior**
Twig comments (`{# ... #}`) should be treated as complete units, with their content ignored during parsing. Any template syntax tokens inside comments should not be parsed as actual template expressions.

**Actual Behavior**
The parser throws: `Parsing error: Expecting "#}", but found "{%" at (line, column)`

**To Reproduce**
Minimal example:
```twig
{# {% include "template.twig" %} #}
```
ESLint Configuration:
```js
// Same as https://github.com/yeonjuan/html-eslint/blob/main/packages/parser/lib/template-engine-syntax-preset.js
parserOptions: {
  templateEngineSyntax: {
    "{{": "}}",
    "{%": "%}",
    "{#": "#}"
  }
}
```

**Additional Notes**
This is a common use case when temporarily commenting out template code during development. Other valid examples that would fail:

```twig
{# TODO: {{ variable }} needs to be escaped #}
{# Disabled: {% for item in items %}...{% endfor %} #}
```

Looking at the [template-engine-syntax-preset.js](https://github.com/yeonjuan/html-eslint/blob/main/packages/parser/lib/template-engine-syntax-preset.js), the current implementation appears to treat all delimiter pairs independently:

```js
const TWIG = {
  "{{": "}}",
  "{%": "%}",
  "{#": "#}",
};
```
This doesn't account for the fact that Twig comments should have higher precedence and their content should be completely ignored, including any nested template syntax.

**Suggested Fix**
The parser should recognize when it's inside a comment block and skip all content until the matching closing delimiter, similar to how HTML comments work. Comment delimiters should take precedence over other template syntax.

**Comment by yeonjuan** [comment]
Hi @webrgp. Thank you for your report. It appears there is a bug where nested template syntax is not being properly checked. As you mentioned, it seems necessary to assign a higher priority to “comments delimiter.” . Let me fix it.

**Comment by yeonjuan** [comment]
Hi @webrgp , It's fixed and released in `v0.51.0`

Use preset 

```
import { TEMPLATE_ENGINE_SYNTAX } from "@html-eslint/parser";

 templateEngineSyntax: TEMPLATE_ENGINE_SYNTAX.TWIG
```

or

```
   templateEngineSyntax: [
                  { open: "{{", close: "}}" },
                  { open: "{%", close: "%}" },
                   { open: "{#", close: "#}", isComment: true },
            ]
```
